src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,413 @@
+    "_JsonSerializer": ("sagemaker.predictor",),
+    "_NpySerializer": ("sagemaker.predictor",),
+    "_CsvDeserializer": ("sagemaker.predictor",),
+    "BytesDeserializer": ("sagemaker.predictor",),
+    "StringDeserializer": ("sagemaker.predictor",),
+    "StreamDeserializer": ("sagemaker.predictor",),
+    "_NumpyDeserializer": ("sagemaker.predictor",),
+    "_JsonDeserializer": ("sagemaker.predictor",),
+    "numpy_to_record_serializer": ("sagemaker.amazon.common",),
+    "record_deserializer": ("sagemaker.amazon.common",),
+}
+
+# The values are tuples so that the object can be passed to matching.matches_any.
+NEW_CLASS_NAME_TO_NAMESPACES = {
+    "CSVSerializer": ("sagemaker.serializers",),
+    "JSONSerializer": ("sagemaker.serializers",),
+    "NumpySerializer": ("sagemaker.serializers",),
+    "CSVDeserializer": ("sagemaker.deserializers",),
+    "BytesDeserializer": ("sagemaker.deserializers",),
+    "StringDeserializer": ("sagemaker.deserializers",),
+    "StreamDeserializer": ("sagemaker.deserializers",),
+    "NumpyDeserializer": ("sagemaker.deserializers",),
+    "JSONDeserializer": ("sagemaker.deserializers",),
+    "RecordSerializer ": ("sagemaker.amazon.common",),
+    "RecordDeserializer": ("sagemaker.amazon.common",),
+}
+
+OLD_CLASS_NAME_TO_NEW_CLASS_NAME = {
+    "_CsvSerializer": "CSVSerializer",
+    "_JsonSerializer": "JSONSerializer",
+    "_NpySerializer": "NumpySerializer",
+    "_CsvDeserializer": "CSVDeserializer",
+    "BytesDeserializer": "BytesDeserializer",
+    "StringDeserializer": "StringDeserializer",
+    "StreamDeserializer": "StreamDeserializer",
+    "_NumpyDeserializer": "NumpyDeserializer",
+    "_JsonDeserializer": "JSONDeserializer",
+    "numpy_to_record_serializer": "RecordSerializer",
+    "record_deserializer": "RecordDeserializer",
+}
+
+OLD_OBJECT_NAME_TO_NEW_CLASS_NAME = {
+    "csv_serializer": "CSVSerializer",
+    "json_serializer": "JSONSerializer",
+    "npy_serializer": "NumpySerializer",
+    "csv_deserializer": "CSVDeserializer",
+    "json_deserializer": "JSONDeserializer",
+    "numpy_deserializer": "NumpyDeserializer",
+}
+
+OLD_AMAZON_CLASS_NAMES = set(
+    {
+        class_name
+        for class_name, namespaces in OLD_CLASS_NAME_TO_NAMESPACES.items()
+        if "sagemaker.amazon.common" in namespaces
+    }
+)
+NEW_AMAZON_CLASS_NAMES = set(
+    {
+        class_name
+        for class_name, namespaces in NEW_CLASS_NAME_TO_NAMESPACES.items()
+        if "sagemaker.amazon.common" in namespaces
+    }
+)
+
+NEW_CLASS_NAMES = set(OLD_CLASS_NAME_TO_NEW_CLASS_NAME.values())
+OLD_CLASS_NAMES = set(OLD_CLASS_NAME_TO_NEW_CLASS_NAME.keys())
+
+OLD_OBJECT_NAMES = set(OLD_OBJECT_NAME_TO_NEW_CLASS_NAME.keys())
+
+
+class SerdeConstructorRenamer(Modifier):
+    """A class to rename SerDe classes."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Call`` node instantiates a SerDe class.
+
+        This looks for the following calls:
```

Comment:
you might want to note that it also looks for those calls without the namespace

Reply:
It should right now. The unit tests for that method tests both calls with namespaces and without.

```
@pytest.mark.parametrize(
    "src, expected",
    [
        ("sagemaker.predictor._CsvSerializer()", True),
        ("sagemaker.predictor._JsonSerializer()", True),
        ("sagemaker.predictor._NpySerializer()", True),
        ("sagemaker.predictor._CsvDeserializer()", True),
        ("sagemaker.predictor.BytesDeserializer()", True),
        ("sagemaker.predictor.StringDeserializer()", True),
        ("sagemaker.predictor.StreamDeserializer()", True),
        ("sagemaker.predictor._NumpyDeserializer()", True),
        ("sagemaker.predictor._JsonDeserializer()", True),
        ("sagemaker.predictor.OtherClass()", False),
        ("sagemaker.amazon.common.numpy_to_record_serializer()", True),
        ("sagemaker.amazon.common.record_deserializer()", True),
        ("_CsvSerializer()", True),
        ("_JsonSerializer()", True),
        ("_NpySerializer()", True),
        ("_CsvDeserializer()", True),
        ("BytesDeserializer()", True),
        ("StringDeserializer()", True),
        ("StreamDeserializer()", True),
        ("_NumpyDeserializer()", True),
        ("_JsonDeserializer()", True),
        ("numpy_to_record_serializer()", True),
        ("record_deserializer()", True),
        ("OtherClass()", False),
    ],
)
```